### DISCUSSION TITLE
Add timezone as user preference per site instead being a General setting

### DISCUSSION THREAD

**Discussion by bzin** [craftcms/cms#8518] [Ideas] [answered]
### Description

Having multiple sites for different countries and languages gives the need of having different timezone for the site itself.

This should/could be added as a user preference instead of having only one timezone per installation. When this would not be defined, timezone could drop to the default timezone set. 

However, this does only solve the problem when the country has only a time zone. Australia, United States and India do have multiple time zones. 

Maybe could also be an idea to include a site preference where you associate your site with a specific country. Therefore it would "automatically" associate multiple time zones to the site id.

**Comment by benjaminkohl** [comment]
This is a feature that has come up for us numerous times over the years for content administrators (CP access users) that live in different timezones than the default timezone and they want to see the date/time field values in the CP adjusted to match where they live.

**Comment by proimage** [comment]
This really needs to be implemented. At the very least, it would be good to have a site setting of "Always show timezone for dates" so that even if the time shown isn't accurate to the current user, they'll at least be informed about it.

**Comment by spork-philip** [comment]
This would be a great feature - is there a timeline on when an 'Enhancement' could be looked into as a 'Feature'?

**Comment by brandonkelly** [comment]
We discussed this today.

There are two distinct paths we could take here:

1. Give each site its own time zone setting, which defaults to the system time zone. All timestamps/time inputs pertaining to that site would use the site’s time zone.
2. Give each user their own “Time Zone” preference, which defaults to the system time zone. All timestamps/time inputs would use the user’s time zone.

(These are mutually exclusive.)

I’m leaning toward option 1, as I think it’s helpful for all users to be on the same page about what time content is being published, etc., with a shared established time zone.

> At the very least, it would be good to have a site setting of "Always show timezone for dates" so that even if the time shown isn't accurate to the current user, they'll at least be informed about it.

Yes—probably not even with a setting; any time we are showing times within Craft, we should be establishing which time zone it’s in, as a static label beside it (e.g. `PST`).

I don’t think giving each user a time zone preference _per site_ makes a lot of sense… both for the same reason I’m leaning toward option 1, and because it would be overkill to expect users to set 5 or 10 different time zone settings.

**Reply by proimage** [reply]
How about this:

- Have a default site timezone, which is the default on the frontend (overrideable by template code), and is the default for backend users.
- Backend users can override the displayed timezone in their user preferences, on a per-installation (not per-multisite) basis. User preference is backend-only; on the frontend, the system time and template code determine the output.
- Whichever timezone is shown to backend users in listings, also have a tooltip that shows the time in UTC—and if the user customized their time, also show the site-default time:

Eg. for a site in Cali and a user in NY:
```
In the backend:

Created at:  Jan 1, 2026 @ 8:00pm [UTC-5] (user)
Tooltip:
- Jan 2, 2026 @ 1:00am [UTC]
- Jan 1, 2026 @ 5:00pm [UTC-8] (system)

On the frontend:
"Article published on Jan 1, 2026 @ 5:00pm [UTC-8]" (which is the system time)
```

**Reply by brandonkelly** [reply]
Yeah so basically, option 2.

I agree that the front end should not be affected by their choice.

**Comment by charliegrinsted** [comment]
Option 1 is the best choice for our upcoming use case but as some of the previous comments here lean towards, there is definitely a case for Option 2 where the focus is on the user location rather than the site location.

Is an order of preference of [user setting] -> [site setting] -> [system setting] not feasible?

**Reply by brandonkelly** [reply]
> Is an order of preference of [user setting] -> [site setting] -> [system setting] not feasible?

I think that would cause too much confusion.

**Comment by brandonkelly** [comment, marked as answer]
Added a new “Time Zone” user preference for Craft 5.10 (5d5f28b8358077a11f614ed1a10e9fc8a42a123e).

**Reply by brandonkelly** [reply]
Related, timestamps in the control panel will start showing their time zone abbreviation (#18639).

**Reply by brandonkelly** [reply]
Craft 5.10.0 is out with that change.

**Reply by alexanderloewe** [reply]
❤️
